internal/services/groups/group_resource.go

```diff
@@ -1153,6 +1211,21 @@ func groupResourceRead(ctx context.Context, d *schema.ResourceData, meta interfa
 	}
 	tf.Set(d, "members", members)
 
+	auId := d.Get("administrative_unit_id")
+	if auId.(string) != "" {
+		isMemberOfAu, _, err := isGroupInAdministrativeUnit(ctx, client, auId.(string), *group.ID())
+		if err != nil {
+			return tf.ErrorDiagPathF(err, "administrativeUnit", "Could not retrieve administrative Unit where this group belongs to %q", d.Id())
+		}
+		if isMemberOfAu {
+			tf.Set(d, "administrative_unit_id", auId)
+		} else {
+			tf.Set(d, "administrative_unit_id", nil)
+		}
+	} else {
+		tf.Set(d, "administrative_unit_id", nil)
+	}
```

Comment:
As above, with this I think the [`/groups/{id}/memberOf`](https://learn.microsoft.com/en-us/graph/api/group-list-memberof?view=graph-rest-1.0) endpoint is a better fit here. In the event that multiple AUs are found, we should sort by object ID and populate with the first one; this should avoid a persistent diff.

Reply:
In the following scenario:
```
resource "azuread_group" "group" {
  ...
  administrative_unit_id = "bcd"
}
```
then adding the group to a second AU with object ID `abc` (e.g. via Portal), this would result in the provider reading in the wrong object_id.

This also relates to the other comment on whether `administrative_unit_id` should be multiple IDs instead.

Reply:
As above, let's make this property a list of AUs `administrative_unit_ids` and absolve ourselves of this problem :)